newlib/libc/search/qsort.c

```diff
@@ -70,6 +70,12 @@ PORTABILITY
 #define inline
 #endif
 
+#ifdef __CHERI__
```

Comment:
We generally use `#if __has_feature(capabilities)` nowadays, but maybe newlib doesn't have the non-clang fallback defined? In that case #fidef __CHERI__ is fine.

Reply:
Yeah, I also followed the convention in mem*() family in newlib; they use `#ifdef __CHERI__`

Reply:
Most of the newlib code dates from when we still had a lot of `__CHERI__` in CheriBSD (or at least if it was changing I was not yet aware of that) so I'm to blame for the ones that are already there. newlib's sys/cdefs.h does have a `__has_feature` fallback as upstream actually uses `__has_feature(nullability)`, and both cheric.h and sys/_stdint.h use `__has_feature(capabilities)`.

Reply:
I changed that to use `__has_feature(capabilities)`